src/lib/route-builder.js

```diff
@@ -0,0 +1,162 @@
+'use strict'
+const _ = require('lodash')
+const uuid = require('uuid4')
+const BigNumber = require('bignumber.js')
+const AssetsNotTradedError = require('../errors/assets-not-traded-error')
+const UnacceptableAmountError = require('../errors/unacceptable-amount-error')
+
+class RouteBuilder {
+  /**
+   * @param {RoutingTables} routingTables
+   * @param {PrecisionCache} precisionCache
+   * @param {Object} config
+   * @param {Integer} config.minMessageWindow seconds
+   * @param {Number} config.slippage
+   * @param {Object} config.ledgerCredentials
+   */
+  constructor (routingTables, precisionCache, config) {
+    this.baseURI = routingTables.baseURI
+    this.routingTables = routingTables
+    this.precisionCache = precisionCache
+    this.minMessageWindow = config.minMessageWindow
+    this.slippage = config.slippage
+    this.ledgerCredentials = config.ledgerCredentials
+  }
+
+  /**
+   * @param {Object} query
+   * @param {String} query.sourceLedger
+   * @param {String} query.sourceAccount
+   * @param {String} query.sourceAmount
+   * @param {String} query.destinationLedger
+   * @param {String} query.destinationAccount
+   * @param {String} query.destinationAmount
+   * @returns {Transfer}
+   */
+  * getQuote (query) {
+    const _nextHop = this._findNextHop(query)
+    if (!_nextHop) throwAssetsNotTradedError()
+    if (query.sourceAmount) {
+      _nextHop.finalAmount = (+_nextHop.finalAmount * (1 - this.slippage)).toFixed(15)
```

Comment:
15 was because according to [bignumber.js](http://mikemcl.github.io/bignumber.js/#bignumber):
> Values of type number with more than 15 significant digits are considered invalid 

I'll change it to just use bignumber for the math, though.